include/highfive/H5DataSet.hpp

```diff
@@ -31,6 +31,11 @@ class DataSet : public Object,
 
     const static ObjectType type = ObjectType::Dataset;
 
+    ///
+    /// \brief return the path to the current dataset
+    /// \return the path to the dataset
+    std::string getPath() const;
+
```

Comment:
Move to `Object`: `DataSet`, `Group`, and `Attribute` derive from it: all would have this functionality